models/image.py

```diff
@@ -152,6 +155,22 @@ def new_image(self):
             return None
         return image[0]
 
+    @new_image.setter
+    def new_image(self, value):
+        if value is None:
+            raise ValueError('Do not assign None to new_image. Simply clear the upstream_images list.')
+        if not isinstance(value, Image):
+            raise ValueError("The new_image must be an Image object.")
+        if len(self.upstream_images) != 2:
+            raise ValueError("This only works for subtractions with exactly two upstream images.")
+
+        if self.upstream_images[0].id == self.ref_image_id:
+            self.upstream_images[1] = value
+        elif self.upstream_images[1].id == self.ref_image_id:
+            self.upstream_images[0] = value
```

Comment:
Did we pick a convention where if there are two upstream images, the first one is the ref and the second one is the new (or vice versa)?  Or did we decide that the upstream images are unordered, because we have ref_image_id and new_image_id?  Based on this code, I'm assuming we did the latter.

Reply:
The `upstream_images` are sorted by MJD so you cannot assume anything on the order of ref vs. new.